cluster-operator/src/main/java/io/strimzi/operator/cluster/model/cruisecontrol/Capacity.java

```diff
@@ -122,26 +129,65 @@ public class Capacity {
     private static final String BROKER_ID_KEY = "brokerId";
     public static final String CAPACITY_KEY = "capacity";
     public static final String DISK_KEY = "DISK";
-    private static final String CPU_KEY = "CPU";
+    public static final String CPU_KEY = "CPU";
     private static final String INBOUND_NETWORK_KEY = "NW_IN";
     private static final String OUTBOUND_NETWORK_KEY = "NW_OUT";
     private static final String DOC_KEY = "doc";
 
-    private final int replicas;
+    public static final String RESOURCE_TYPE = "cpu";
+
     private final Storage storage;
 
-    public Capacity(KafkaSpec spec, Storage storage) {
-        io.strimzi.api.kafka.model.balancing.BrokerCapacity bc = spec.getCruiseControl().getBrokerCapacity();
+    private enum ResourceRequirementType {
+        REQUEST,
+        LIMIT
+    }
 
+    public Capacity(KafkaSpec spec, Storage storage) {
         this.capacityEntries = new TreeMap<>();
-        this.replicas = spec.getKafka().getReplicas();
         this.storage = storage;
 
-        processCapacityEntries(bc);
+        processCapacityEntries(spec);
+    }
+
+    private static Integer getResourceRequirement(ResourceRequirements resources, ResourceRequirementType requirementType) {
+        if (resources != null) {
+            Map<String, Quantity> resourceRequirement = requirementType == ResourceRequirementType.REQUEST ? resources.getRequests() : resources.getLimits();
+            if (resourceRequirement != null) {
+                Quantity quantity = resourceRequirement.get(RESOURCE_TYPE);
```

Comment:
In fact you put this logic in the enum too and have `Quantity quantity = requirementType.getQuantity(resources)`

Reply:
@tomncooper Can you double check to make sure my refactoring makes sense?

I do want to be fancy